source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc

```diff
@@ -46,6 +54,8 @@ void ProxyFilter::onDestroy() {
 }
 
 Http::FilterHeadersStatus ProxyFilter::decodeHeaders(Http::RequestHeaderMap& headers, bool) {
+  latchTime(decoder_callbacks_, dnsStart());
```

Comment:
I'm curious about latching the time here vs. latching at other points in this function. For instance, only if the filter gets to `loadDnsCacheEntry`?

Reply:
yeah it's the difference between when we start the lookup and when we start doing "DNS work"
I can go either way - moved.